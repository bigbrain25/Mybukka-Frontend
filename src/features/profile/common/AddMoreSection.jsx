import React from 'react';

import PropTypes from 'prop-types';

import './addMoreSection.scss';

const AddMoreSection = ({ onClick, children, text, dataTarget }) => (
  <div
    className="add-more-section"
    data-target={dataTarget}
    onClick={onClick}
    aria-pressed="false"
    tabIndex="0"
    role="button"
    data-toggle="modal"
  >
    {children}
    <span className="text-add">{text}</span>
  </div>
);

export default AddMoreSection;

AddMoreSection.defaultProps = {
  dataTarget: ''
};

AddMoreSection.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.arrayOf(PropTypes.node),
    PropTypes.node
  ]).isRequired,
  text: PropTypes.string.isRequired,
  dataTarget: PropTypes.string
};
