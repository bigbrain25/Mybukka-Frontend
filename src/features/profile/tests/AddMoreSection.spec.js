import React from 'react';

import AddMoreSection from '../common/AddMoreSection';

describe('AddMoreSection component', () => {
  const props = {
    dataTarget: '#modal',
    text: 'add new card'
  };
  const wrapper = mount(
    <AddMoreSection {...props}>
      <p>child</p>
    </AddMoreSection>
  );

  it('renders properly', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
