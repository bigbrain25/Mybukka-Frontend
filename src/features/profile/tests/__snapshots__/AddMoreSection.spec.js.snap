// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddMoreSection component renders properly 1`] = `ReactWrapper {}`;
